### PR TITLE
Docs/hdi

### PR DESCRIPTION
**Pull request by jost-s**
### Summary

- Docs: replace references to deprecated macros `entry_def!` and `#[hdk_entry(id..)]`

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [x] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

File: crates/hdk/src/lib.rs [hunk 1]
```diff
@@ -331,7 +331,17 @@ pub mod countersigning;
 /// Most Holochain applications will define their own app entry types.
 ///
 /// App entries are all entries that are not system entries.
-/// They are defined in the `entry_defs` callback and then the application can call CRUD functions with them.
+/// Definitions of entry types belong in the integrity zomes of a DNA. In contrast, operations
```
Comment (jost-s) [inline crates/hdk/src/lib.rs:334]: @maackle Is there a way to dedup these doc comments? They need to be updated in two different places, the hdk and the hdi crates.

**Review by maackle — CHANGES_REQUESTED** [review]
Looks like most instances of example snippets using `hdk_entry_helper` need to include an `hdk_entry_defs` enum as well

File: crates/hdk/src/entry.rs [hunk 2]
```diff
@@ -129,8 +139,13 @@ where
...
-/// struct Foo(u32);
+/// #[hdk_entry_helper]
+/// pub struct Foo(u32);
+///
+/// #[hdk_entry_defs]
+/// pub enum EntryTypes {
+///    Foo(Foo)
+/// }
 ///
 /// let foo_zero_action_hash: ActionHash = commit_entry!(Foo(0))?;
```
Comment (maackle) [inline crates/hdk/src/entry.rs:150]: I think commit_entry! is actually outdated, which I didn't catch the first time around.

```suggestion
/// let foo_zero_action_hash: ActionHash = create_entry(EntryTypes::Foo(Foo(0)))?;
```

**Review by maackle — DISMISSED** [review]
A few more issues

**Comment by jost-s** [comment]
@maackle Thanks for spotting that, I was sloppy there. This should be it now.

File: crates/hdk/src/entry.rs [hunk 3]
```diff
@@ -129,11 +139,16 @@ where
...
+/// pub struct Foo(u32);
+///
+/// #[hdk_entry_defs]
+/// pub enum EntryTypes {
+///    Foo(Foo)
+/// }
 ///
-/// let foo_zero_action_hash: ActionHash = commit_entry!(Foo(0))?;
-/// let foo_ten_update_action_hash: ActionHash = update_entry(foo_zero_action_hash, Foo(10))?;
+/// let foo_zero_action_hash: ActionHash = create_entry!(EntryTypes::Foo(Foo(0)))?;
```
Comment (maackle) [inline crates/hdk/src/entry.rs:150]: It's not a macro

```suggestion
/// let foo_zero_action_hash: ActionHash = create_entry(EntryTypes::Foo(Foo(0)))?;
```

**Review by maackle — COMMENTED** [review]
I think it's good now! Thanks for the updates

**Review by maackle — APPROVED** [review]

**Pull request merged**